Source/Helpers/LogHelper.cs

```diff
@@ -72,24 +72,26 @@ public static void WriteToLog()
 		const string LogFileName = "Log.txt";
 		StringBuilder log = new();
 		lock (Logs)
+		{
```

Comment:
Instead of locking the whole method (and making logging slower) use a synchronized `TextWriter`, you can then safely flush asynchronously.
[MSDocs](https://learn.microsoft.com/en-us/dotnet/api/system.io.textwriter.synchronized?view=net-8.0&redirectedfrom=MSDN#System_IO_TextWriter_Synchronized_System_IO_TextWriter_)